sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -347,6 +347,16 @@ querySpecification
        windows?)
     ;
 
+hint
+    : '/*+' hintStatement '*/'
+    ;
+
+hintStatement
+    : hintName=identifier
+    | hintName=identifier '(' parameters+=identifier parameters+=identifier ')'
```

Comment:
Thank you for review, @cloud-fan .
The first goal of this PR provides a general syntax for hints, not only broadcast hints. The `(` and `)` syntax is for `INDEX(t idx_emp)` style. You can see the testcase for this in the testcase, too.